File: mobile/android/VehicleStatusModel.java
package com.autoconnect.android;

/** Overall status of a vehicle shown on the dashboard. */
public class VehicleStatusModel {
    public int vehicleId;
    public String make;
    public String model;
    public int year;
    public int mileage;
    public String vehicleType; // ICE | EV | HYBRID

    public VehicleStatusModel(int vehicleId, String make, String model,
                              int year, int mileage, String vehicleType) {
        this.vehicleId = vehicleId;
        this.make = make;
        this.model = model;
        this.year = year;
        this.mileage = mileage;
        this.vehicleType = vehicleType;
    }

    public String displayName() {
        return year + " " + make + " " + model;
    }
}

File: mobile/ios/VehicleStatusModel.swift
import Foundation

/// Overall status of a vehicle shown on the dashboard.
struct VehicleStatusModel: Codable, Identifiable {
    let id: Int
    let vin: String
    let make: String
    let model: String
    let year: Int
    let mileage: Int
    let vehicleType: String  // ICE | EV | HYBRID

    enum CodingKeys: String, CodingKey {
        case id, vin, make, model, year, mileage
        case vehicleType = "vehicle_type"
    }

    var displayName: String { "\(year) \(make) \(model)" }
}
